CPS-XXXXX/README.md

```diff
@@ -0,0 +1,71 @@
+---
+CPS: ?
+Title: Query Layer Standardization
+Status: Open
+Category: Tools
+Authors:
+    - Vladimir Kalnitsky <vladimir@mlabs.city>
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/625
+    - `#wallet-connectors` in CIP editors Discord: https://discord.gg/tkp6r6ESJR
+Created: 2023-11-27
+---
+
+## Abstract
+
+Cardano lacks a standardized query layer. This leads to suboptimal tooling, dApp and wallet architecture.
+
+## Problem
+
+### Query Layers and Tooling
+
+Lack of a standardized query layer results in multiple different implementations of roughly the same set of functionality:
+
+- [Blockfrost](https://blockfrost.io/)
+- [Koios](https://www.koios.rest/)
+- [Maestro](https://www.gomaestro.org)
+- [Ogmios](https://ogmios.dev/)
+
+As a result, there is a need to support multiple incompatible APIs in downstream tools, examples of which are:
+
+- [Mesh.js](https://meshjs.dev/providers)
+- [Lucid](https://lucid.spacebudz.io/)
+- [cardano-transaction-library](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/runtime.md)
+
+Query layer providers are not identical, which means that the *promise* of abstracting away from a particular query layer provider completely, that an offchain library may want to give to its users, will either be left *unfulfilled* (i.e. some features will work with some providers, but not others) or the scope of the downstream API will have to be *reduced* to the very minimum that is covered by every supported query layer.
+
+### Query Layers and Wallets
+
+This CPS initiative originated in the discussion about [Extensive Wallet Standard CIP](https://github.com/cardano-foundation/CIPs/pull/620) in [Discord](https://discord.com/channels/971785110770831360/992011119872970762/1176567729017327737) ([invite](https://discord.gg/P59aNVN8zu)).
+
+Every light wallet has its own backend infrastructure: functioning of the browser extension relies on the availability of the data sources. However, none of the wallets currently provide a way to override their query layer endpoint URLs.
+
+In the past, users have encountered problems with ability to submit transactions due to limited mempool capacity, which resulted in some wallets providing a way to configure custom [tx-submit-api](https://github.com/blinklabs-io/tx-submit-api) endpoints - but this only covers transaction submission.
+
+Tight coupling between wallets and query layers results in unnecessary economic burden imposed on wallets, and forces them to make opinionated choices (of query layer provider) that could be delegated to the end user, similarly to how it is done in Metamask with RPC provider selection.
```

Comment:
```suggestion
Tight coupling between wallets and query layers results in unnecessary economic burden imposed on wallets, and forces them to make opinionated choices (of query layer provider) that could otherwise be delegated to the end user, similarly to how it is done in Metamask with RPC provider selection.
```